### ISSUE TITLE
[fsd50k] make handling of multipart zip work across unix systems

### ISSUE THREAD

**Issue by iranroman** [soundata/soundata#55]
The `zip` subprocess in `soundata/datasets/fsd50k.py` uses the syntax of the zip utility found in MacOS. As a result, the tests fail in other unix systems (i.e. Ubuntu, Archlinux). See the issue in [https://unix.stackexchange.com/questions/40480/how-to-unzip-a-multipart-spanned-zip-on-linux](https://unix.stackexchange.com/questions/40480/how-to-unzip-a-multipart-spanned-zip-on-linux)

Ideal solution: avoid using python `subprocess` and instead carry out the operation exclusively with native python libraries.

**Comment by justinsalamon** [comment]
@genisplaja we could look at using: https://docs.python.org/3/library/zipfile.html#module-zipfile

**Comment by genisplaja** [comment]
Thanks @iranroman for catching the issue and @justinsalamon for the suggestion. I actually took a look at `zipfile` while implementing the loader for FSD50K to translate the merging commands into python library functions. However, I could not find a workaround using `zipfile` to properly merge the split zip files.

Probably this is not the best solution but we could detect the running system using `platform.system()`. If output is `Darwin` we run the command as it is, and if the output is `Linux` we run the following (as suggested in the post attached by @iranroman):
```
cat FSD50K.dev_audio.z* > ~/unzip_dev.zip
```
In any case, I'll experiment a bit more with `zipfile`. I agree that moving to native python libraries for these processes would be ideal!

**Comment by genisplaja** [comment]
Hello! Thanks @iranroman for raising the issue and @justinsalamon for the suggestion. I have been experimenting with `zipfile` quite a bit and it doesn't look like there is a workaround for the necessary commands to merge the FSD50K zip splits. I've got in touch with E. Fonseca and he successfully tested the commands in an Ubuntu machine. Now @magdalenafuentes is also testing it! 
Even though I understand that the ideal case would be to avoid running commands through `subprocess`. I will try again soon to review this issue to see if there's a workaround!

**Comment by iranroman** [comment]
Hello all, Thanks for your attention to this issue. Perhaps just use `DOWNLOAD_INFO` in `fsd50k.py` to describe the nature of the zip files, so that users can merge the files as they prefer and better fits their OS?

**Comment by genisplaja** [comment]
So we would throw the `DOWNLOAD_INFO` message and we would download the files automatically without merging? Maybe we could also have a `_download()` and a `_download_merge_unzip()` so users can chose whether they merge the files at convencience or they try to do it automatically with the function provided by us. But I don't know if that complicates things a bit...!

**Comment by iranroman** [comment]
I have implemented handling of multipart zip files in #62 . This handles download, merging, and unzipping of multipart zip files in `download_utils.py`. 

The way it works is allowing `REMOTES` keys to have a list of `download_utils.RemoteFileMetadata`. If a `list`, then it is a multipart zip file and it is merged and unzipped as such. 

This resulted in removal of three `fsd50` tests related to download, merging, unzipping and cleanup. Instead, I'm using the functions already implemented in `download_utils.py` (see the git diff). All tests passed before I created the PR, including the very long one `pytest -s tests/test_full_dataset.py --local --dataset fsd50`. 

This implementation will also be helpful for more datasets I will add (see PR #62)

> Maybe we could also have a _download() and a _download_merge_unzip() so users can chose whether they merge the files at convencience or they try to do it automatically with the function provided by us.

It is unpractical for users to get only some of these files and not all at once. `unzip` will throw an error message due to the zip file not being complete. All parts of each multipart zip file must be downloaded, merged, and unzipped.
